openpaas-james/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/openpaas/james/common/LinagoraKeystoreSetMethodContract.scala

```diff
@@ -0,0 +1,489 @@
+package com.linagora.openpaas.james.common
+
+import com.linagora.openpaas.james.common.LinagoraKeystoreSetMethodContract.{PGP_KEY, PGP_KEY_ARMORED, PGP_KEY_ID}
+import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
+import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.http.ContentType.JSON
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import org.apache.commons.codec.binary.Base64
+import org.apache.http.HttpStatus
+import org.apache.http.HttpStatus.SC_OK
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.core.ResponseObject.SESSION_STATE
+import org.apache.james.jmap.http.UserCredential
+import org.apache.james.jmap.rfc8621.contract.Fixture.{ACCEPT_RFC8621_VERSION_HEADER, ACCOUNT_ID, ALICE_ACCOUNT_ID, BOB, BOB_BASIC_AUTH_HEADER, BOB_PASSWORD, DOMAIN, authScheme, baseRequestSpecBuilder, getHeadersWith}
+import org.apache.james.jmap.rfc8621.contract.tags.CategoryTags
+import org.apache.james.utils.DataProbeImpl
+import org.junit.jupiter.api.{BeforeEach, Disabled, Tag, Test}
+
+import java.nio.charset.StandardCharsets
+
+object LinagoraKeystoreSetMethodContract {
+  private val PGP_KEY: Array[Byte] = ClassLoader.getSystemClassLoader
+    .getResourceAsStream("gpg.pub")
+    .readAllBytes()
+
+  private val PGP_KEY_ARMORED: String = new String(PGP_KEY, StandardCharsets.UTF_8)
+    .replace("\n", "\\n")
+
+  private val PGP_KEY_ID: String = "3BA423385C8C80D453D7E6F95BF4E866E9CC6FA2"
+}
+
+trait LinagoraKeystoreSetMethodContract {
+
+  @BeforeEach
+  def setUp(server : GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString(), BOB_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .build()
+  }
+
+  @Test
+  def keystoreSetCreateShouldSucceedWithValidKey(): Unit = {
+    val request = s"""{
+                     |	"using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:pgp"],
+                     |	"methodCalls": [
+                     |		["Keystore/set", {
+                     |			"accountId": "$ACCOUNT_ID",
+                     |			"create": {
+                     |        "K87": {
+                     |          "key": "$PGP_KEY_ARMORED"
+                     |        }
+                     |      }
+                     |		}, "c1"]
+                     |	]
+                     |}""".stripMargin
```

Comment:
weird indent here?

Reply:
and many other places...

Reply:
weird... Fine locally... with intellij and visual code... not sure if it is github or my IDEs being crazy here oO